clang/test/CodeGenHLSL/builtins/dot2add.hlsl

```diff
@@ -0,0 +1,78 @@
+// RUN: %clang_cc1 -finclude-default-header -fnative-half-type -triple \
+// RUN:   dxil-pc-shadermodel6.3-compute %s -emit-llvm -disable-llvm-passes -o - | \

```

Comment:
remove `-disable-llvm-passes` here and on line 5. The hlsl intrinsics aren't inlining and so these tests are not confirming the code you wrote. You will want to see an intrinsic for dot2add in the DXIL case and you will want to see a dot product intrinsic followed by a multiply in the SPIRV case.